Removal of `COAP_MUTEX_DEFINE()` breaks Zephyr build

## Environment

- Build System:             CMake (via Zephyr `west`)
- Operating System:         Linux
- Operating System Version: Ubuntu 24.04
- Hosted Environment:       Zephyr

## libcoap Configuration Summary

Copy the "libcoap Configuration Summary" output lines here after running
`./configure` (with any options that you have specified)
  or
`cmake ..` (with any -D options that you have specified)
  or if neither of the above two methods is used (do from within the libcoap directory)
`git describe --tags`

## Problem Description

Seeing the following build error(s):
```
[329/398] Building C object modules/golioth-firmware-sdk/CMakeFiles/..__port__zephyr.dir/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c.obj
In file included from /home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_internal.h:111,
                 from /home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c:16:
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c: In function 'coap_show_pdu':
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c:732:19: warning: passing argument 1 of 'sys_mutex_lock' from incompatible pointer type [-Wincompatible-pointer-types]
  732 |   coap_mutex_lock(&m_show_pdu);
      |                   ^~~~~~~~~~~
      |                   |
      |                   coap_mutex_t * {aka struct sys_mutex_t *}
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_mutex_internal.h:92:46: note: in definition of macro 'coap_mutex_lock'
   92 | #define coap_mutex_lock(a)    sys_mutex_lock(a, K_FOREVER)
      |                                              ^
In file included from /home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_mutex_internal.h:86:
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/zephyr/include/zephyr/sys/mutex.h:151:52: note: expected 'struct sys_mutex *' but argument is of type 'coap_mutex_t *' {aka 'struct sys_mutex_t *'}
  151 | static inline int sys_mutex_lock(struct sys_mutex *mutex, k_timeout_t timeout)
      |                                  ~~~~~~~~~~~~~~~~~~^~~~~
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c:1048:21: warning: passing argument 1 of 'sys_mutex_unlock' from incompatible pointer type [-Wincompatible-pointer-types]
 1048 |   coap_mutex_unlock(&m_show_pdu);
      |                     ^~~~~~~~~~~
      |                     |
      |                     coap_mutex_t * {aka struct sys_mutex_t *}
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_mutex_internal.h:94:48: note: in definition of macro 'coap_mutex_unlock'
   94 | #define coap_mutex_unlock(a)  sys_mutex_unlock(a)
      |                                                ^
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/zephyr/include/zephyr/sys/mutex.h:156:54: note: expected 'struct sys_mutex *' but argument is of type 'coap_mutex_t *' {aka 'struct sys_mutex_t *'}
  156 | static inline int sys_mutex_unlock(struct sys_mutex *mutex)
      |                                    ~~~~~~~~~~~~~~~~~~^~~~~
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c: In function 'coap_log_impl':
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c:1199:21: warning: passing argument 1 of 'sys_mutex_lock' from incompatible pointer type [-Wincompatible-pointer-types]
 1199 |     coap_mutex_lock(&m_log_impl);
      |                     ^~~~~~~~~~~
      |                     |
      |                     coap_mutex_t * {aka struct sys_mutex_t *}
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_mutex_internal.h:92:46: note: in definition of macro 'coap_mutex_lock'
   92 | #define coap_mutex_lock(a)    sys_mutex_lock(a, K_FOREVER)
      |                                              ^
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/zephyr/include/zephyr/sys/mutex.h:151:52: note: expected 'struct sys_mutex *' but argument is of type 'coap_mutex_t *' {aka 'struct sys_mutex_t *'}
  151 | static inline int sys_mutex_lock(struct sys_mutex *mutex, k_timeout_t timeout)
      |                                  ~~~~~~~~~~~~~~~~~~^~~~~
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/external/libcoap/src/coap_debug.c:1206:23: warning: passing argument 1 of 'sys_mutex_unlock' from incompatible pointer type [-Wincompatible-pointer-types]
 1206 |     coap_mutex_unlock(&m_log_impl);
      |                       ^~~~~~~~~~~
      |                       |
      |                       coap_mutex_t * {aka struct sys_mutex_t *}
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/port/zephyr/../../external/libcoap/include/coap3/coap_mutex_internal.h:94:48: note: in definition of macro 'coap_mutex_unlock'
   94 | #define coap_mutex_unlock(a)  sys_mutex_unlock(a)
      |                                                ^
/home/hasheddan/code/github.com/golioth/golioth-firmware-sdk/zephyr/include/zephyr/sys/mutex.h:156:54: note: expected 'struct sys_mutex *' but argument is of type 'coap_mutex_t *' {aka 'struct sys_mutex_t *'}
  156 | static inline int sys_mutex_unlock(struct sys_mutex *mutex)
```

This is broken after https://github.com/obgm/libcoap/pull/1197, but I was able to fix with the following patch:
```
diff --git a/include/coap3/coap_mutex_internal.h b/include/coap3/coap_mutex_internal.h
index f14d2fdc..c9358499 100644
--- a/include/coap3/coap_mutex_internal.h
+++ b/include/coap3/coap_mutex_internal.h
@@ -85,7 +85,7 @@ typedef int coap_mutex_t;
 #elif defined(__ZEPHYR__)
 #include <zephyr/sys/mutex.h>
 
-typedef struct k_mutex coap_mutex_t;
+typedef struct sys_mutex coap_mutex_t;
 
 #define coap_mutex_init(a)    sys_mutex_init(a)
 #define coap_mutex_destroy(a)
```

I will open a PR to implement 👍🏻 

### Expected Behavior

I expected for compilation to be successful.

### Actual Behavior

Compilation failed with steps described above.

### Steps to reproduce

`west build` with https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/zephyr/golioth_basics

--- a/include/coap3/coap_mutex_internal.h
+++ b/include/coap3/coap_mutex_internal.h
@@ -85,7 +85,7 @@ typedef int coap_mutex_t;
 #elif defined(__ZEPHYR__)
 #include <zephyr/sys/mutex.h>
 
-typedef struct k_mutex coap_mutex_t;
+typedef struct sys_mutex coap_mutex_t;
 
 #define coap_mutex_init(a)    sys_mutex_init(a)
 #define coap_mutex_destroy(a)

